### DISCUSSION TITLE
Generate STL file in NodeJS

### DISCUSSION THREAD

**Discussion by jvlobo** [microsoft/maker.js#610] [Q&A] [answered]
Hello!!

I just found out about maker.js and I think it is great.

I'm trying to generate an STL file using NodeJS but I'm having some trouble.
This is my code, I copied part of it from an example somewhere (that was done for web) and adapt it to the Node code:

```js
const makerjs = require('makerjs');
const { CAG } = require('@jscad/csg');
const stlSerializer = require('@jscad/stl-serializer');
const fs = require('fs');

const star = new makerjs.models.Star(7, 100, 50);
const svg = makerjs.exporter.toSVG(star);

// save svg to file
fs.writeFileSync('star.svg', svg);

const stl = makerjs.exporter.toJscadSTL(CAG, stlSerializer, star, { extrude: 50 });

// save stl to file
fs.writeFileSync('star.stl', stl);

```

The SVG file is generated correctly, I can open the SVG and I see the star drawing.
However, in the STL generation, I get this error in the console:

>node_modules/@jscad/stl-serializer/index.js:59
  if (objects3d.length === 0) throw new Error('only 3D geometries can be serialized to STL')
  
 Any ideas of what I may be doing wrong?
 
 Thanks!!

**Comment by danmarshall** [comment, marked as answer]
Hello, sorry but the Jscad implementation in Maker.js is way out of date. (The versions on the website are old frozen copies). The Jscad project has matured to the level of importing SVG, so you are probably best off with exporting to SVG then bringing that into Jscad's [svg deserializer](https://jscad.app/docs/module-io_svg-deserializer.html).
